### PR TITLE
Episode Detail

### PR DESCRIPTION
**Pull request by juanca172**
Simulator Screen Recording - iPhone 14 Pro Max - 2023-09-04 at 22 53 48

### CODE REVIEW

**Comment by hgp691** [comment]
Please merge first https://github.com/SSPF20/RickAndMortyAPI/pull/23 to access the Hosting Controller

File: RickAndMorty/Extensions/String+Ext.swift [hunk 1]
```diff
@@ -18,6 +18,14 @@ extension String {
         return nil
     }
     
+    var pageNumber: Int? {
```
Comment (hgp691) [inline RickAndMorty/Extensions/String+Ext.swift]: No longer needed, check the new Property Wrapper Approach

File: RickAndMorty/Views/RMDetailEpisode/DetailEpisodeViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,32 @@
+//
+//  DetailEpisodeViewModel.swift
+//  RickAndMorty
+//
+//  Created by Juan Camilo Fonseca Gomez on 3/09/23.
+//
+
+import Foundation
+import Observation
```
Comment (hgp691) [inline RickAndMorty/Views/RMDetailEpisode/DetailEpisodeViewModel.swift]: Double Check the Observation API, You no longer need to add ObservableObject conformation, Now the approach is to add a Property Wrapper

File: RickAndMorty/Views/RMDetailEpisode/RMDetailEpisode.swift [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+struct RMDetailEpisode: View {
+    
+    var viewModel : DetailEpisodeViewModel
+    
+    var body: some View {
+        VStack (alignment: .center, spacing: 10) {
+            HStack {
+                VStack(alignment: .leading,spacing: 10) {
+                    Text(NSLocalizedString("Name: \(viewModel.name)", comment: ""))
```
Comment (hgp691) [inline RickAndMorty/Views/RMDetailEpisode/RMDetailEpisode.swift:18]: Move the entire logic for text creation to the View Model

File: RickAndMorty/Views/RMListView/RMListViewController.swift [hunk 4]
```diff
@@ -100,6 +105,17 @@ final class RMListViewController<A: Decodable, B: Configuration>: UIViewControll
             }
     }
     
+    //extension
```
Comment (hgp691) [inline RickAndMorty/Views/RMListView/RMListViewController.swift]: ??

File: RickAndMorty/Extensions/String+Ext.swift (hunk 1 above)
Comment (juanca172) [reply RickAndMorty/Extensions/String+Ext.swift]: Ready, waiting for review.

File: RickAndMorty/Views/RMListView/RMListViewController.swift (hunk 4 above)
Comment (juanca172) [reply RickAndMorty/Views/RMListView/RMListViewController.swift]: Already removed.

File: RickAndMorty/Views/RMDetailEpisode/DetailEpisodeViewModel.swift (hunk 2 above)
Comment (juanca172) [reply RickAndMorty/Views/RMDetailEpisode/DetailEpisodeViewModel.swift]: I removed the import and conformation

**Review by hgp691 — APPROVED** [review]

**Pull request merged**